### PR TITLE
Add View History

### PR DESCRIPTION
**Pull request by bastianrickmann**

### CODE REVIEW

File: ui/src/app/components/user-page/user-page.component.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { GalleryItem, IframeItem, ImageItem, VideoItem } from 'ng-gallery';
 import { Title } from '@angular/platform-browser';
 import { MatSnackBar } from '@angular/material/snack-bar';
 import { HttpErrorResponse } from "@angular/common/http";
+import {ViewHistoryService} from "../../services/view-history.service";
```
Comment (alantoch) [inline ui/src/app/components/user-page/user-page.component.ts]: This needs some formatting.

File: ui/src/app/services/content-retriever.service.ts [hunk 2]
```diff
@@ -44,4 +45,15 @@ export class ContentRetrieverService {
...
   }
 
+  public getMultipleUserAboutData(users: string[]): Observable<UserAboutData[]> {
+    return of(...users).pipe(
+      map(user => this.getUserAboutDetails(user)),
+      catchError((error, caught) => {
+        console.log(error)
+        return of()}),
+      flatMap(user => user),
+      tap(user => console.log(user)),
```
Comment (alantoch) [inline ui/src/app/services/content-retriever.service.ts]: Do we want to log the users to the console?

File: ui/src/app/services/view-history.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    history.add(username);
+    localStorage.setItem("history", JSON.stringify(Array.from(history)));
+  }
+
+  public getUserViewHistory(): string[] {
+    const userHistory: string[] = JSON.parse(localStorage.getItem("history") || "[]");
+    return userHistory;
+  }
+
+  public removeUserFromHistory(username: string): void {
```
Comment (alantoch) [inline ui/src/app/services/view-history.service.ts:25]: Are we using this function anywhere?

**Review by alantoch — COMMENTED** [review]
We should go through the files and format them. Alternatively there's a VS code setting that formats the file when it's saved.

File: ui/src/app/services/view-history.service.ts (hunk 3 above)
Comment (bastianrickmann) [reply ui/src/app/services/view-history.service.ts:25]: now, we do.

**Pull request merged**